src/base/static/components/story-sidebar/index.js

```diff
@@ -0,0 +1,190 @@
+import React, { Component } from "react";
+import PropTypes from "prop-types";
+import { Map, OrderedMap, fromJS } from "immutable";
+
+import StoryChapter from "./story-chapter";
+import constants from "../../constants";
+
+import "./index.scss";
+
+import { story as storyConfig, place as placeConfig } from "config";
+
+class StorySidebar extends Component {
+  constructor(props) {
+    super(props);
+
+    this.stories = Object.keys(storyConfig).reduce((memo, storyName) => {
+      return memo.set(
+        storyName,
+        Object.keys(storyConfig[storyName].order).reduce((memo, chapterUrl) => {
+          return memo.set(
+            chapterUrl,
+            fromJS(this.getModelFromUrl(chapterUrl).attributes),
+          );
+        }, OrderedMap()),
+      );
+    }, OrderedMap());
+
+    this.state = {
+      currentStoryHeader: "",
+      currentStoryDescription: "",
+      currentStoryName: "",
+      currentStoryChapters: OrderedMap(),
+      currentStoryRoute: "",
+      currentChapter: Map(),
+    };
+  }
+
+  componentWillMount() {
+    this.isInitialized = false;
+    this.loadAndSetStoryChapter(Backbone.history.getFragment().split("/"));
+    this.isInitialized = true;
+
+    this.props.router.on("route", (fn, route) => {
+      this.loadAndSetStoryChapter(route);
+    });
+  }
+
+  loadAndSetStoryChapter(route) {
+    let isRouteFound = false;
+    const joinedRoute = route.join("/");
+    this.stories.forEach((story, storyName) => {
+      if (story.get(joinedRoute)) {
+        // If the current route exists in this story, load the story and set
+        // the current chapter and name.
+        isRouteFound = true;
+        this.setState({
```

Comment:
We should refactor this to be a `find` instead of a `forEach` to make it clear that `setState` won't get called multiple times.

Also, I think it would be more clear if `this.stories.find` returns the state object itself, and if no route was found, we'd set it to the default values below. Then we would return that state object, and call `setState` in `componentDidMount`, making it easier to track any side effects. We can rename this method `getCurrentStoryState`.